cura/Settings/MachineManager.py

```diff
@@ -1496,3 +783,268 @@ def setSettingForAllExtruders(self, setting_name: str, property_name: str, prope
         for key, extruder in self._global_container_stack.extruders.items():
             container = extruder.userChanges
             container.setProperty(setting_name, property_name, property_value)
+
+    #
+    # New
+    #
+
+    # We not fetch it from _current_root_material_id, but later we can get it from somewhere else
+    @pyqtProperty("QVariantList", notify = rootMaterialChanged)
+    def currentExtruderPositions(self):
+        return sorted(list(self._current_root_material_id.keys()))
+
+    @pyqtProperty("QVariant", notify = rootMaterialChanged)
+    def currentRootMaterialId(self):
+        # initial filling the current_root_material_id
+        self._current_root_material_id = {}
+        for position in self._global_container_stack.extruders:
+            self._current_root_material_id[position] = self._global_container_stack.extruders[position].material.getMetaDataEntry("base_file")
+        return self._current_root_material_id
+
+    @pyqtProperty("QVariant", notify = rootMaterialChanged)
```

Comment:
Same concern as `currentRootMaterialId`: We're depending on QML to update `self._current_root_material_name`. If `currentRootMaterialName` is not used in QML, the Python field won't get updated.

Reply:
Same as above, will try to fix this later.

Reply:
Fixed in a87db2d721e7779c4b6d3650aa60155e0de22ea4.